engine/src/main/java/org/camunda/bpm/engine/impl/interceptor/ProcessDataContext.java

```diff
@@ -144,6 +141,7 @@ public boolean pushSection(ExecutionEntity execution) {
 
     addToStack(activityIdStack, execution.getActivityId());
     addToStack(execution.getProcessDefinitionId(), mdcPropertyDefinitionId);
+    addToStack(execution.getProcessDefinition().getKey(), mdcPropertyDefinitionKey);
```

Comment:
Compared to the static values like `processDefitionId`, fetching the process definition can be a costly operation at specific points in time. This requires parsing the whole process definition from XML into memory, for example. Have you considered this here already and thought about disabling this by default, as we do for the business key?